File: 2012 Book Source Code/Chapter_7_Source_Code/Chapter 7/Employee Class/Employee.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Security.Permissions;
using System.Data;


namespace Pro_SSRS
{
	public class Employee
	{
		public Employee()
		{
		}
		[PermissionSetAttribute(SecurityAction.Assert, Unrestricted = true)]
		public static decimal CostPerVisitXML(string employeeID, DateTime visitDate)
		{

			DataSet empDS = new DataSet();
			empDS.ReadXmlSchema(@"C:\Temp\EmployeePay.xsd");
			empDS.ReadXml(@"C:\Temp\EmployeePay.xml");
			DataRow[] empRows = empDS.Tables["EmployeePay"].Select("EmployeeID = '" + employeeID + "'");
			Decimal empAmt;
			if (empRows.Length > 0)
			{
				empAmt = Convert.ToDecimal(empRows[0]["Amount"]);
				return empAmt;
			}
			else
				return 0;

		}

		[PermissionSetAttribute(SecurityAction.Assert, Unrestricted = true)]
		public static decimal CostPerVisitWS(string employeeID, DateTime visitDate)
		{

			EmployeeWS.Employee empWS = new EmployeeWS.Employee();
			Decimal empAmt;
			empAmt = empWS.GetCost(employeeID, visitDate);
			return empAmt;

		}
	}
}

File: 2012 Book Source Code/Chapter_7_Source_Code/Chapter 7/Employee VB/Employee.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Security.Permissions
Imports System.Data

Public Class Employee

    <PermissionSetAttribute(SecurityAction.Assert, Unrestricted:=True)>
    Public Shared Function CostPerVisitXML(ByVal employeeID As String, ByVal visitDate As DateTime) As Decimal
        Dim empDS As New DataSet()

        empDS.ReadXmlSchema("C:\Temp\EmployeePay.xsd")
        empDS.ReadXml("C:\Temp\EmployeePay.xml")

        Dim empRows() As DataRow = empDS.Tables("EmployeePay").Select("EmployeeID = '" + employeeID + "'")
        Dim empAmt As Decimal

        If (empRows.Length > 0) Then
            empAmt = Convert.ToDecimal(empRows(0)("Amount"))
            Return empAmt
        Else
            Return 0
        End If

    End Function

    <PermissionSetAttribute(SecurityAction.Assert, Unrestricted:=True)>
    Public Shared Function CostPerVisitWS(ByVal employeeID As String, ByVal visitDate As DateTime) As Decimal
        Dim empWS As New EmployeeWS.Employee()
        Dim empAmt As Decimal

        empAmt = empWS.GetCost(employeeID, visitDate)
        Return empAmt
    End Function



End Class
